# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.cpp
# Commit: 85b827971a48
# Full Hash: 85b827971a48b6eca0c765fc649b6cc2275134eb
# Author: Yulia <ystartsev@mozilla.com>
# Date: 2022-12-15 09:27:59
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Call EnsureAndLoadStringBundle from Worker ScriptLoaderRunnable; r=allstarschh
#   
#   Depends on D163238
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D163239
# ==============================================================================

diff -r d647be139836 -r 85b827971a48 dom/workers/ScriptLoader.cpp
--- a/dom/workers/ScriptLoader.cpp	Wed Dec 14 14:31:20 2022 +0000
+++ b/dom/workers/ScriptLoader.cpp	Wed Dec 14 14:55:33 2022 +0000
@@ -1009,6 +1009,9 @@
 nsresult ScriptLoaderRunnable::Run() {
   AssertIsOnMainThread();
 
+  // Ensure the string bundle is loaded, in case the ModuleLoader errors out.
+  nsContentUtils::EnsureAndLoadStringBundle(nsContentUtils::eDOM_PROPERTIES);
+
   // Convert the origin stack to JSON (which must be done on the main
   // thread) explicitly, so that we can use the stack to notify the net
   // monitor about every script we load. We do this, rather than pass
